src/components/ui/time-slider/time-slider.tsx

```diff
@@ -0,0 +1,141 @@
+import { observer } from "mobx-react-lite";
+import React, { useEffect, useState } from "react";
+import MapSlider from "../../../assets/icons/map-slider.svg";
+import PauseIcon from "../../../assets/icons/pause-icon.svg";
+import PlayIcon from "../../../assets/icons/play-icon.svg";
+import TimeSliderThumb from "../../../assets/icons/time-slider-thumb.svg";
+import { formatDateString } from "../../../utilities/date-utils";
+import { UIButton } from "../ui-button";
+import { UIButtonContainer } from "../ui-button-container";
+import { DateRangeSliderThumb } from "./date-range-slider-thumb";
+import { SliderThumb } from "./slider-thumb";
+import { TimeLine } from "./time-line";
+import {
+  labelHeight, labelOffsets, mapSliderThumbOffset, timeLineTop, timeSliderThumbOffset
+} from "./time-slider-contsants";
+import "./time-slider.scss";
+
+const labelBaseTop = timeLineTop - labelHeight / 2;
+let lastFrameTime = Date.now();
+const animationRate = 0.1;
+let animationTimeout = 0;
+
+interface ITimeSliderProps {
+  dateMax: number;
+  dateMin: number;
+}
+export const TimeSlider = observer(function TimeSlider({ dateMax, dateMin }: ITimeSliderProps) {
+  const [upperRangePercent, setUpperRangePercent] = useState(1);
+  const [lowerRangePercent, setLowerRangePercent] = useState(0);
+  const [mapPercent, setMapPercent] = useState(0);
+  const [timePercent, setTimePercent] = useState(1);
+  const [animating, setAnimating] = useState(false);
+
+  useEffect(() => {
+    setMapPercent(value => Math.min(upperRangePercent, Math.max(lowerRangePercent, value)));
+    setTimePercent(value => Math.min(upperRangePercent, Math.max(lowerRangePercent, value)));
+  }, [upperRangePercent, lowerRangePercent]);
+
+  useEffect(() => {
+    if (animating && !animationTimeout) {
+      animationTimeout = setTimeout(() => {
+        animationTimeout = 0;
+        const now = Date.now();
+        const delta = (now - lastFrameTime) / 1000;
+        lastFrameTime = now;
+        const newMapPercent = Math.min(timePercent + animationRate * delta, upperRangePercent);
+        setTimePercent(newMapPercent);
+        if (newMapPercent >= upperRangePercent) {
+          setAnimating(false);
+        }
+      });
```

Comment:
It is interesting that the setTimeout doesn't have any delay. So I'd guess the slider just redraws as fast as it can. Doing that is not very efficient. I'd recommend using `requestAnimationFrame` instead. I think you could just replace `setTimeout` with it and this could will still behave the same way.

Reply:
I tried switching this to use `requestAnimationFrame`, but I was having trouble because it uses react state. I'm going to leave this as is for now, but when I hook this into the models so I can use observable state I'll give it another try. It's too bad I can't just use `useFrame` here!